kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -294,11 +306,17 @@ protected void writeAndFlushResponseToClient(Channel channel) {
                         response.getRequest().getHeader());
                 }
 
-                ByteBuf result = responseToByteBuf(response.getResponseFuture().get(), response.getRequest());
+                ByteBuf result = responseToByteBuf(response.getResponseFuture().get(
+                        kafkaConfig.getRequestTimeoutMs(),
+                        TimeUnit.MILLISECONDS), response.getRequest());
                 channel.writeAndFlush(result);
-            } catch (Exception e) {
-                // should not comes here.
-                log.error("error to get Response ByteBuf:", e);
+            } catch (ExecutionException | TimeoutException | InterruptedException e) {
+                // get response result may timeout.
+                log.error("request {}: error to get Response ByteBuf: ",
+                        response.getRequest().getHeader().apiKey(), e.getCause());
```

Comment:
Maybe I didn't express well before. We should use `getCause()` only for `ExecutionException`, because it's thrown  from `CompletableFuture.exceptionally` and `getCause()` is the real cause. For `TimeoutException` and `InterruptedException`, `getCause()` returns null.